ipaserver/plugins/host.py

```diff
@@ -871,7 +871,9 @@ def pre_callback(self, ldap, dn, *keys, **options):
                 )
 
         if self.api.Command.ca_is_enabled()['result']:
-            certs = self.api.Command.cert_find(host=keys)['result']
+            certs = self.api.Command.cert_find(
+                subject=keys[-1], status='VALID'
```

Comment:
If the user called `ipa host-del host1` but the fqdn is `host1.ipa.test`, `keys[-1]` is `host1` and the filter `subject=host1` may match other hosts. I agree that host-del will also remove them but I think it's safer to use a filter based on fqdn that is evaluated a few lines above: cert_find(`subject=fqdn`,...).

Reply:
I'll take a look, thanks.

Reply:
Nice catch, PR updated and rebased.